cli-messaging/src/main/java/com/salesforce/messaging/CliMessager.java

```diff
@@ -43,34 +54,59 @@ public void addMessage(MessagePassableException exception) {
 	 * @param eventKey    EventKey to display to user
 	 * @param args        String args passed to the EventKey to make the displayed message meaningful
 	 */
+    @Deprecated
 	public void addMessage(String internalLog, EventKey eventKey, String... args) {
-		// Developer error if eventKey was not added to exception and we'll get a bunch of NPEs
-		assert (eventKey != null);
-		// Confirm that the correct number of arguments for the message has been provided
-		// If this fails, this would be a developer error
-		assert (eventKey.getArgCount() == args.length);
-
-		final Message message = new Message(
-			eventKey.getMessageKey(),
-			Arrays.asList(args),
-			internalLog,
-			eventKey.getMessageType(),
-			eventKey.getMessageHandler(),
-			eventKey.isVerbose());
-		MESSAGES.add(message);
+        final Message message = createMessage(internalLog, eventKey, args);
+        MESSAGES.add(message);
 	}
 
-	/**
+    /**
+     * Publish formatted stdout message to pass onto Typescript layer.
+     * Make sure EventKey is updated with messages/EventKeyTemplates.json
+     * and has correct properties in the enum.
+     *
+     * @param internalLog Information for internal use. Will be logged but not displayed to user
+     * @param eventKey    EventKey to display to user
+     * @param args        String args passed to the EventKey to make the displayed message meaningful
+     */
+    public static void postMessage(String internalLog, EventKey eventKey, String... args) {
```

Comment:
Creating an alternate method to post realtime messages. I've marked the other entry points as deprecated and we can eventually make them realtime as well.